core/src/main/java/io/dekorate/kubernetes/decorator/DeploymentResourceFactory.java

```diff
@@ -39,9 +40,10 @@ public HasMetadata create(AbstractKubernetesManifestGenerator<?> generator, Base
         .endMetadata()
         .withNewSpec()
         .withReplicas(1)
-        .withNewTemplate()
```

Comment:
I disagree with this change. 

Our decorators should work regardless of the structure of the initial resource (users are allowed to bring their own resource and can't possibly know the decorate internals).

When we have a decorator that requires the exististence of some objects, we should then ensure that these objects exists (usually by also adding the decorators that create them).

So, instead of creating the template and metadata here, we should be using the decorators that do so.